src/main/java/seedu/address/model/appointment/Appointment.java

```diff
@@ -81,6 +81,15 @@ public boolean hasConflict(Appointment toCheck) {
                 && getTimeslot().hasOverlap(toCheck.getTimeslot());
     }
 
+    /**
+     * Returns whether there is a scheduling conflict between this appointment and the other.
+     */
+    public boolean hasEditConflict(Appointment toCheck) {
+        return (patient.equals(toCheck.getPatient())
```

Comment:
This should be replaced with
`(patient.equals(toCheck.getPatient())
                || doctor.equals(toCheck.getDoctor()))
                && getTimeslot().hasOverlap(toCheck.getTimeslot());`

Reply:
agree,
rationale is that for a scheduling conflict to happen, either patient has overlapping timeslot, or doctor has overlapping timeslot,

think this method wouldn't work, and `hasConflict` should be used instead

Reply:
agree, thanks for pointing out